ui/numbered_menu.py

```diff
@@ -0,0 +1,49 @@
+from ui import Menu
+from ui.base_list_ui import BaseListUIElement
+from ui.utils import clamp
+
+
+class UIWidget(object):
+    pass  # todo : complete
+
+
+class NumberedMenu(UIWidget, Menu):
+    """
+    todo: options:
+    - activate entry after type or select only
+    - delay for keypresses
+    """
+
+    def __init__(self, *args, **kwargs):
+        Menu.__init__(self, *args, **kwargs)
+        self.key_map = {
```

Comment:
Not a very descriptive variable name, especially considering there's already self.keymap IIRC.